File: packages/server/src/adapters/logger/logger.ts
import { logger } from "../../logger";
import { ILogger } from "../../ports";

export class Logger implements ILogger {
  error(msg: string) {
    logger.error(msg);
  }
  warn(msg: string) {
    logger.warn(msg);
  }
  info(msg: string) {
    logger.info(msg);
  }
  verbose(msg: string) {
    logger.verbose(msg);
  }
  debug(msg: string) {
    logger.debug(msg);
  }
  silly(msg: string) {
    logger.silly(msg);
  }
}

File: server-scala/src/main/scala/net/kgtkr/anontown/ports/Logger.scala
package net.kgtkr.anontown.ports;
import zio.IO
import net.kgtkr.anontown.AtServerError

trait Logger {
  def error(msg: String): IO[AtServerError, Unit];
  def warn(msg: String): IO[AtServerError, Unit];
  def info(msg: String): IO[AtServerError, Unit];
  def verbose(msg: String): IO[AtServerError, Unit];
  def debug(msg: String): IO[AtServerError, Unit];
  def silly(msg: String): IO[AtServerError, Unit];
}

trait LoggerComponent {
  val logger: Logger;
}
